test/resources/test_list.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module MailchimpAPI
+  class TestList < Test::Unit::TestCase
+    def setup
+      super
+
+      stub_request(:get, 'https://__api_region_identifier__.api.mailchimp.com/3.0/lists')
+        .to_return body: load_fixture(:lists)
+    end
+
+    def test_countable
+      stub_request(:get, 'https://__api_region_identifier__.api.mailchimp.com/3.0/lists?count=0&fields=total_items')
+        .to_return body: load_fixture(:count_payload)
+
+      assert_equal 2, MailchimpAPI::List.count
+    end
+
+    def test_instantiates_proper_class
+      lists = MailchimpAPI::List.all
+
+      assert_kind_of MailchimpAPI::CollectionParsers::List, lists
+      assert_instance_of MailchimpAPI::List, lists.first
+    end
+
+    def test_fetches_all_lists
+      lists = MailchimpAPI::List.all
+
+      assert_equal 2, lists.count
+      assert_equal 2, lists.total_items
+    end
+
+    def test_lists_has_valid_links
+      lists = MailchimpAPI::List.all
+
+      assert_kind_of MailchimpAPI::Link, lists.links.sample
+    end
+
+    def test_fetches_specific_list
+      list_id = '6574d0bcc7'
+
+      stub_request(:get, "https://__api_region_identifier__.api.mailchimp.com/3.0/lists/#{list_id}")
```

Comment:
Any reason why you didn't use Addressable::Template and then calling `.expand` on the parameters you want to swap in?